src/ci/docker/host-x86_64/rfl/Dockerfile

```diff
@@ -0,0 +1,56 @@
+FROM ubuntu:23.10
+
+ARG DEBIAN_FRONTEND=noninteractive
+
+RUN apt-get update && apt-get install -y --no-install-recommends \
+  g++ \
+  gcc-multilib \
+  make \
+  ninja-build \
+  file \
+  curl \
+  ca-certificates \
+  python3 \
+  git \
+  cmake \
+  gdb \
+  flex \
+  bison \
+  bc \
+  llvm-17 \
+  llvm-17-dev \
+  llvm-17-tools \
+  clang-17 \
+  lld-17 \
+  libelf-dev \
+  libedit-dev \
+  libssl-dev \
+  pkg-config \
+  zlib1g-dev \
+  xz-utils \
+  && rm -rf /var/lib/apt/lists/*
+
+COPY scripts/sccache.sh /scripts/
+RUN sh /scripts/sccache.sh
+
+# RfL needs access to llvm-ar
+ENV PATH="${PATH}:/usr/lib/llvm-17/bin"
+
+# We are disabling CI LLVM since this builder is intentionally using a host
+# LLVM, rather than the typical src/llvm-project LLVM.
+ENV NO_DOWNLOAD_CI_LLVM 1
+ENV EXTERNAL_LLVM 1
+
+# This is not the latest LLVM version, so some components required by tests may
+# be missing.
+ENV IS_NOT_LATEST_LLVM 1
+
+# Using llvm-link-shared due to libffi issues -- see #34486
+ENV RUST_CONFIGURE_ARGS \
+      --build=x86_64-unknown-linux-gnu \
+      --llvm-root=/usr/lib/llvm-17 \
+      --enable-llvm-link-shared \
+      --set rust.thin-lto-import-instr-limit=10
```

Comment:
It looks like this is largely copied from the llvm-17 builder. Can we make note of that? I wonder if we should be reusing the builder/Dockerfile somehow...

Reply:
It is indeed copied from there. I plan to separate the Docker images from the scripts that we run in CI scripts, but that is a non-trivial refactoring, I'd prefer not to block this PR on that.

Reply:
It sounds like if we switch to download-ci-llvm or sccache, we'd avoid most of the copying, right?

Reply:
Well, we'll still need to copy-paste *some* Dockerfile, since we don't really have a way to share Dockerfiles at the moment, right?

Reply:
Well, yes, but at least there's less going on. e.g., src/ci/docker/host-x86_64/x86_64-gnu/Dockerfile looks "simpler" to me and the stuff there that's shared is already copied into lots of files. So I think it would be simpler, and also doesn't need edits as we continue advancing the llvm versions we support.

Reply:
Ok, fair enough, I'll reuse some simpler Dockerfile then.

Reply:
Although I remember now that the RfL build did actually require some extra tools that weren't even in the original LLVM 17 Dockerfile, so it probaly won't be much shorter.